testing/testing.gni

```diff
@@ -156,7 +156,13 @@ template("dart_snapshot") {
   testonly = true
 
   dart_snapshot_kernel_target_name = "_dsk_$target_name"
-  dart_snapshot_kernel_path = "$target_gen_dir/assets/kernel_blob.bin"
+
+  if (defined(invoker.dart_kernel)) {
+    dart_snapshot_kernel_path = invoker.dart_kernel
```

Comment:
Instead of threading this through as a parameter, can you just always prepend the target name to the file name like:
```
dart_snapshot_kernel_path= "$target_gen_dir/assets/${target_name}_kernel_blob.bin"
```

Reply:
the target name has the prefix `_ds` at this point. I reworked this a bit, so I added a flag `use_target_as_artifact_prefix` which is set in the root target to just put the root's target name as prefix.